server/src/test/java/org/elasticsearch/common/xcontent/BaseXContentTestCase.java

```diff
@@ -947,11 +948,18 @@ void doTestRawValue(XContent source) throws Exception {
             assertNull(parser.nextToken());
         }
 
-        os = new ByteArrayOutputStream();
+        final AtomicBoolean closed = new AtomicBoolean(false);
+        os = new ByteArrayOutputStream() {
+            @Override
+            public void close() {
+                closed.set(true);
+            }
+        };
         try (XContentGenerator generator = xcontentType().xContent().createGenerator(os)) {
             generator.writeStartObject();
             generator.writeFieldName("test");
             generator.writeRawValue(new BytesArray(rawData).streamInput(), source.type());
+            assertFalse(closed.get());
```

Comment:
What do you think about adding a message here? e.g.

```suggestion
            assertFalse("Generator should not have close the output stream", closed.get());
```

Reply:
++